base/doc/clsguide.tex

```diff
@@ -977,6 +977,30 @@ \subsection{Passing options around}
 handler, but this handler is not used for |landscape| because that option is
 explicitly declared.
 
+\subsection{Testing dates of other code}
+
+\begin{decl}
+  |\IfPackageAtLeast(TF)| \arg{package} \arg{date} \arg{true code} \arg{false code}\\
+  |\IfClassAtLeast(TF)| \arg{class} \arg{date} \arg{true code} \arg{false code}\\
+  |\IfFileAtLeast(TF)| \arg{file} \arg{date} \arg{true code} \arg{false code}
+\end{decl}
+There are occasions when you need to test if a package is
+at least a particular date (assuming it has been loaded). This can be done using the
```

Comment:
Worth mentioning that `\IfPackageAtLeastTF` always uses the false branch if the specified package is not loaded (yet)?

Reply:
Possibly - but it would be odd to test the version without knowing it's loaded

Reply:
I can't recall details of how I encountered such odd uses before. Just guessing a less odd case: when the package name was misspelled.

Also I find this is the special case of saying "`\IfPackageAtLeastTF` never errors." Invalid date format leaves true branch, package not-loaded (yet) leaves false branch. Maybe these should be classified as undefined behavior.

Reply:
invalid date means the date becomes an small number typically zero so ends up being earlier than any reasonable date, hence the results depending on what you ask for. But I agree these cases should be listed as undefined behavior.